lib/Onyx.js

```diff
@@ -1606,6 +1625,58 @@ function update(data) {
         }
     });
 
+    const promises = [];
+
+    // Group all the collection-related keys and update each collection in a single `mergeCollection` call
+    onyxCollectionKeySet.forEach((collectionKey) => {
+        const collectionItemKeys = _.filter(_.keys(updateQueue), (key) => isKeyMatch(collectionKey, key));
+        if (_.size(collectionItemKeys) <= 1) {
+            // If there are no items of this collection in the updateQueue, we should skip it.
+            // If there is only one item, we should update it individually, therefore retain it in the updateQueue.
+            return;
+        }
+
+        const batchedCollectionUpdates = _.reduce(
+            collectionItemKeys,
+            (acc, key) => {
+                const operations = updateQueue[key];
+
+                // Remove the collection-related key from the updateQueue so that it won't be processed individually later.
+                delete updateQueue[key];
+
+                const updatedValue = applyMerge(undefined, operations, false);
+                if (_.includes(operations, null)) {
+                    acc.set[key] = updatedValue;
+                } else {
+                    acc.merge[key] = updatedValue;
+                }
+                return acc;
+            },
+            {
+                merge: {},
+                set: {},
+            },
+        );
+
+        if (!_.isEmpty(batchedCollectionUpdates.merge)) {
+            promises.push(() => mergeCollection(collectionKey, batchedCollectionUpdates.merge));
+        }
+        if (!_.isEmpty(batchedCollectionUpdates.set)) {
+            promises.push(() => multiSet(batchedCollectionUpdates.set));
+        }
+    });
+
+    _.each(updateQueue, (operations, key) => {
+        const batchedChanges = applyMerge(undefined, operations, false);
+
+        const shouldOverwriteExistingValue = _.includes(operations, null);
+        if (shouldOverwriteExistingValue) {
+            promises.push(() => set(key, batchedChanges));
+        } else {
+            promises.push(() => merge(key, batchedChanges));
+        }
```

Comment:
It seems possible for `null` values to appear in positions other than index 0, which implies that `merge` operations are allowed to pass `null` as a merge value.

Instead of attempting to identify a `null` value here, wouldn't it be more effective to address it during the `enqueueMergeOperation`? If the value provided for a key is `null`, we could directly modify the update queue for this key to `updateQueue[key] = [null];`, effectively treating the update for the key as a `set` operation since the initial value would be overridden by `null`.

This approach would enable us to eliminate irrelevant merges that precede the `null` value, and also simplify the check here having to iterate all operations looking for `null`

Reply:
I speculate that the decision to forego `multiSet` and `multiMerge` methods was made to capitalize on the existing logic for notifying subscribers and updating cache. 

It's still possible to try and leverage the `multi` methods in a potential follow-up